src/python/pants/base/exception_sink_test.py

```diff
@@ -1,128 +1,127 @@
-
-            # We only logged a single error, so the files should both contain only that single log entry.
-            err_rx = """\
+        # We only logged a single error, so the files should both contain only that single log entry.
+        err_rx = f"""\
 timestamp: ([^\n]+)
-process title: {fake_process_title}
+process title: fake_title
 sys.argv: ([^\n]+)
 pid: {pid}
-{msg}
-""".format(
-                fake_process_title=re.escape(fake_process_title), pid=pid, msg=msg
-            )
-            with open(cur_process_error_log_path, "r") as cur_pid_file:
-                self.assertRegex(cur_pid_file.read(), err_rx)
-            with open(shared_error_log_path, "r") as shared_log_file:
-                self.assertRegex(shared_log_file.read(), err_rx)
-
-    def test_backup_logging_on_fatal_error(self):
-        sink = self._gen_sink_subclass()
-        with self.captured_logging(level=logging.ERROR) as captured:
-            with temporary_dir() as tmpdir:
-                sink.reset_log_location(tmpdir)
-                with unittest.mock.patch.object(
-                    sink, "_try_write_with_flush", autospec=sink
-                ) as mock_write:
-                    mock_write.side_effect = ExceptionSink.ExceptionSinkError("fake write failure")
-                    sink._log_exception("XXX")
-        errors = list(captured.errors())
-        self.assertEqual(2, len(errors))
-
-        def format_log_rx(log_file_type):
-            return ".*".join(
-                re.escape(s)
-                for s in [
-                    "pants.base.exception_sink: Error logging the message 'XXX' to the {log_file_type} file "
-                    "handle for".format(log_file_type=log_file_type),
-                    "at pid {pid}".format(pid=os.getpid()),
-                    "\nfake write failure",
-                ]
+XXX
+"""
+        with open(cur_process_error_log_path, "r") as cur_pid_file:
+            assert bool(re.search(err_rx, cur_pid_file.read()))
+        with open(shared_error_log_path, "r") as shared_log_file:
+            assert bool(re.search(err_rx, shared_log_file.read()))
+
+
+def test_backup_logging_on_fatal_error(caplog):
+    sink = _gen_sink_subclass()
+    with temporary_dir() as tmpdir:
+        sink.reset_log_location(tmpdir)
+        with unittest.mock.patch.object(sink, "_try_write_with_flush", autospec=sink) as mock_write:
+            mock_write.side_effect = ExceptionSink.ExceptionSinkError("fake write failure")
+            sink._log_exception("XXX")
+
+    errors = [record for record in caplog.records if record.levelname == "ERROR"]
+    assert len(errors) == 2
+
+    def assert_log(log_file_type: str, log):
+        assert bool(
+            re.search(
+                fr"Error logging the message 'XXX' to the {log_file_type} file handle for .* at "
+                fr"pid {os.getpid()}",
+                log.msg,
             )
+        )
+        assert log.filename == "exception_sink.py"
```

Comment:
The mechanics are a little different than before, but we end up checking the same thing.

Reply:
👍